File: Tema 05 - Typescript/PetShelterBackend/PetShelter.DataAccessLayer/Models/Fundraiser.cs
namespace PetShelter.DataAccessLayer.Models;

public class Fundraiser : IEntity
{
    public int Id { get; set; }
    public string Name { get; set; }
    public decimal GoalValue { get; set; }
    public decimal CurrentRaisedAmount { get; set; }
    public DateTime DueDate { get; set; }
    public DateTime CreationDate { get; set; }
    public string Status { get; set; }
    public int OwnerId { get; set; }
    public Person Owner { get; set; }
    public ICollection<Person> Donors { get; set; }
}

File: Tema 06 - React/petshelter.ui/src/Models/Fundraiser.ts
import type { Person } from "./Person"

export class Fundraiser {
    id: number;
    name: string;
    status: string;
    currentRaisedAmount: number;
    goalValue: number;
    dueDate: Date;
    donors: Person[] | undefined;

    constructor(id: number, name: string, status: string, currentRaisedAmount: number, goalValue: number, dueDate: Date, donors?: Person[]) {
        this.id = id;
        this.name = name;
        this.status = status;
        this.currentRaisedAmount = currentRaisedAmount;
        this.goalValue = goalValue;
        this.dueDate = dueDate;
        this.donors = donors;
    }
}
